core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -966,6 +971,32 @@ public QueryManagerConfig setFaultTolerantExecutionTaskDescriptorStorageMaxMemor
         return this;
     }
 
+    public DataSize getFaultTolerantExecutionTaskDescriptorStorageHighWaterMark()
+    {
+        return faultTolerantExecutionTaskDescriptorStorageHighWaterMark;
+    }
+
+    @Config("fault-tolerant-execution-task-descriptor-storage-high-water-mark")
+    @ConfigDescription("When memory used to store task descriptors for fault tolerant queries is above high water mark we start to compress the storage")
+    public QueryManagerConfig setFaultTolerantExecutionTaskDescriptorStorageHighWaterMark(DataSize faultTolerantExecutionTaskDescriptorStorageHighWaterMark)
+    {
+        this.faultTolerantExecutionTaskDescriptorStorageHighWaterMark = faultTolerantExecutionTaskDescriptorStorageHighWaterMark;
+        return this;
+    }
+
+    public DataSize getFaultTolerantExecutionTaskDescriptorStorageLowWaterMark()
+    {
+        return faultTolerantExecutionTaskDescriptorStorageLowWaterMark;
+    }
+
+    @Config("fault-tolerant-execution-task-descriptor-storage-low-water-mark")
+    @ConfigDescription("When memory used to store task descriptors for fault tolerant queries is below water mark we no longer compress the storage")
+    public QueryManagerConfig setFaultTolerantExecutionTaskDescriptorStorageLowWaterMark(DataSize faultTolerantExecutionTaskDescriptorStorageLowWaterMark)
```

Comment:
Do not compress the storage when tasks descriptors in memory are below given data size

Reply:
🌻